src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2535,8 +2556,8 @@ private void InheritNullableStateOfMember(int targetContainerSlot, int valueCont
                         {
                             return;
                         }
-                        value = this.State.HasValue(valueMemberSlot) ?
-                            this.State[valueMemberSlot] :
+                        value = valueMemberSlot > 0 ?
```

Comment:
Could you help me understand this bit? Is it that HasValue is really a bounds check and the normalizing process might change the size of the state?

Reply:
The main purpose for the change is to remove `HasValue` which was a bounds check. In some places we were doing a bounds check and in others we were doing unchecked access which would initialize which default values...
Now, as long as the slot is > 0, we can get the state and get the adequate default.
That said, this check doesn't currently fail. I'd rather leave it since we had it and it's safer.